plugins/org.eclipse.papyrus.uml.interaction.model/src/org/eclipse/papyrus/uml/interaction/internal/model/spi/impl/DefaultDiagramHelper.java

```diff
@@ -196,6 +197,40 @@ public Shape getLifelineBodyShape(View lifelineView) {
 		return new DeferredCreateCommand<>(Shape.class, editingDomain, parameters, shape);
 	}
 
+	@Override
+	public CreationCommand<Shape> createDestructionOccurrenceShape(Supplier<? extends MessageEnd> destruction,
+			Shape lifelineBody, int yPosition) {
+
+		// TODO: The Logical Model is required to have no dependencies on the diagram editor,
+		// but this is usually the responsibility of a View Provider. That should be okay ...
+
+		Supplier<Shape> shape = () -> {
+			LayoutConstraint llBounds = lifelineBody.getLayoutConstraint();
+			int width = 0;
+			if (Size.class.isInstance(llBounds)) {
+				width = Size.class.cast(llBounds).getWidth();
+			}
+			int execWidth = 20;
+
+			Shape result = NotationFactory.eINSTANCE.createShape();
+			result.setType("Shape_Destruction_Specification");
```

Comment:
Please move this view type into org.eclipse.papyrus.uml.interaction.model.spi.ViewTypes.